stellargraph/datasets/datasets.py

```diff
@@ -707,3 +707,42 @@ def load(self):
             ``graph``.
         """
         return _load_tsv_knowledge_graph(self)
+
+
+class IAEnronEmployees(
+    DatasetLoader,
+    name="IAEnronEmployees",
+    directory_name="ia-enron-employees",
+    url="http://nrvis.com/download/data/dynamic/ia-enron-employees.zip",
+    url_archive_format="zip",
+    expected_files=["ia-enron-employees.edges", "readme.html"],
+    description="A dataset of edges that represent emails sent from one employee to another."
+    "There are 50572 edges, and each of them contains timestamp information. "
+    "Edges refer to 151 unique node IDs in total.",
+    source="http://networkrepository.com/ia-enron-employees.php",
+    url_contains_directory=False,
+):
+    def load(self):
+        """
+        Load this data into a set of nodes and edges
+
+        Returns:
+            A tuple ``(nodes, edges)`` where each element is a Pandas DataFrame. Edges are kept
```

Comment:
I think it would be good to return a `StellarGraph`, as well as potentially the `nodes` and `edges` DataFrames (or maybe just the `edges` one). `StellarGraph` is our native graph format, and so its good to try to phrase things in terms of that. What's the motivation for returning 'raw' dataframes?

Reply:
For now, this dataset is only used for the CTDNE demo, which needs to preprocess the edgelist further to create the graph and train/test examples. I noticed other dataset loaders actually do this also in the `load` method, but I thought it might be good to show how the edges are split in the notebook. What do you think?

Reply:
Yeah, it's probably relevant to show the splitting details in the notebook, and I don't think we have sufficient functionality on `StellarGraph` to do that splitting directly. Maybe we could just add comments for the various bits and pieces here, like the string conversion below and that fact that it doesn't return a `StellarGraph` object, so we don't completely forget the reasoning.

Another option for this would be to return `(graph, edges)` but basically ignore the graph (other than `graph.nodes()`) in the current notebook. This might be good because it's more future compatible: switching `load` to returning `(graph, edges)` (or `(nodes, edges, graph)` or something) would be a breaking change (theoretically), but starting with `(graph, edges)` and converting notebook to use `graph` more would once the `StellarGraph` APIs are available would not be. (That said, I imagine not too many users would use this in their own code.)

Reply:
Returning `(graph, edges)` sounds good. I'll add a comment about the string IDs.